src/hooks/useNavbarToggle.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useRef, useState } from "react"
+
+const useNavbarToggle = () => {
+  const [isToggle, setIsToggle] = useState(false)
+  const toggleRef = useRef<HTMLButtonElement>(null)
+
+  const handleOutSideClick = (e: Event) => {
+    const current = toggleRef.current
+    if (isToggle && current && !current.contains(e.target as Node)) {
+      setIsToggle(false)
+    }
+  }
+
+  const handleToggle = () => {
+    setIsToggle((prevIsToggle: boolean) => !prevIsToggle)
+  }
+
+  useEffect(() => {
+    document.addEventListener("click", handleOutSideClick)
+
+    return () => {
+      document.removeEventListener("click", handleOutSideClick)
+    }
+  }, [isToggle])
```

Comment:
```suggestion
 }, [handleOutSideClick, isToggle])
```

해당 부분 디펜던시 누락된것같습니다! 그리고 위의 handleOutSide 도 useCallback 해주셔야 경고가 안뜰거같습니다